orangecontrib/spectroscopy/widgets/owpolar.py

```diff
@@ -411,6 +411,58 @@ def process_polar_abs(images, alpha, feature, map_x, map_y, invert, polangles, a
 
     return outputs, model, spectra, meta, vars[1]
 
+class PolarDomainContextHandler(DomainContextHandler):
```

Comment:
Could you explain why the new context matcher was needed?

Reply:
I was having an issue where the 'DomainContextHandler' and 'PerfectDomainContextHandler' would save contexts for the multi-in inputs with the last used angle separator for multifile inputs, which wasn't in the domain to begin with. So I would get a keyerror next time the same data was used.

I think this is an ugly solution so I will definitely try and find a better way. I'm sure there is an obvious solution that I a haven't thought of

Reply:
A much better solution in [f9d7d9e](https://github.com/Quasars/orange-spectroscopy/pull/615/commits/f9d7d9e88cead0c87f7ef52618fc45a7201fd6b6).
It turns out that if one processed some data through the widget using the multifile input and then the multi-in input, `self.angles` was keeping the last set value and so would be saved in the context associated with the multi-in data. Subsequent loading of this context would try and put the saved `self.angles` value into the combobox, which would either not have that value in the datas domain, or the combobox would be disabled, hence errors

Reply:
Oh, this is much nicer, thanks!